microcosm_eventsource/immutable_event_factory.py

```diff
@@ -0,0 +1,81 @@
+"""
+Factory for immutable events.
+
+In extension of EventFactory, this class provides a native support of event updating container objects based
+on event types
+
+"""
+from collections import defaultdict
+from typing import Dict
+
+from microcosm_eventsource.factory import EventFactory
+
+
+container_common_mutator_registry: Dict[str, list] = defaultdict(list)
```

Comment:
`Dict[str, List[str]]`

Reply:
Also should these be defined CAP_CASE? I think it makes it's usage later clearer

Reply:
honestly these are internal so I don't care about the naming. Given they are internal and nothing outside this function should touch them, they could probably be `_` prefixed.

Reply:
Agreed. I will update the registry to be _prefixed.